### PR TITLE
Containerize

### PR DESCRIPTION
**Pull request by pravinbhat**
Added dockerfile to do the below
Java 8 based on eclipse-temurin
Spark 2.4.8
Cassandra-Data-Migrator packaged jar
DSBulk
CQLSH

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# Copy CDM jar & template files
+COPY ./target/cassandra-data-migrator-*.jar /assets/
+COPY ./src/resources/sparkConf.properties /assets/
+COPY ./src/resources/partitions.csv /assets/
+COPY ./src/resources/primary_key_rows.csv /assets/
+COPY ./src/resources/runCommands.txt /assets/
+
+# Add all migration tools to path
+ENV PATH="${PATH}:/assets/dsbulk-1.10.0/bin/:/assets/cqlsh-astra/bin/:/assets/spark-2.4.8-bin-hadoop2.7/bin/"
```
Comment (msmygit) [inline Dockerfile:22]: This today will be `dsebulk-1.0.0` but, that version could change. How do we future proof this? 🤔

File: src/resources/runCommands.txt [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 // Random Partitioner Run Command
-spark-submit --properties-file /<path>/sparkConf.properties --verbose --master "local[8]" --conf spark.source.minPartition=-1 --conf spark.source.maxPartition=170141183460469231731687303715884105728 --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-1.x.jar
+spark-submit --properties-file /<path>/sparkConf.properties --verbose --master "local[8]" --conf spark.origin.minPartition=-1 --conf spark.origin.maxPartition=170141183460469231731687303715884105728 --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-1.x.jar
 
-// Diff Data
-spark-submit --properties-file /<path>/sparkConf.properties --verbose --master "local[8]" --class datastax.astra.migrate.DiffData /<path>/cassandra-data-migrator-1.x.jar
+// Validate
+spark-submit --properties-file /<path>/sparkConf.properties --master "local[8]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.DiffData /<path>/cassandra-data-migrator-2.x.jar &> table_out.log
```
Comment (msmygit) [inline src/resources/runCommands.txt:14]: What if someone tries to run this on a system that isn't having enough (i.e. the `25G`) memory free-space? Same for other places too. 🤔

**Comment by msmygit** [comment]
Fixes #43

File: Dockerfile (hunk 1 above)
Comment (pravinbhat) [reply Dockerfile:22]: @msmygit good point, I think we should rename to non-version specific name post untar?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
+FROM eclipse-temurin:8-jammy
+
+RUN apt update && apt install -y openssh-server vim
+RUN service ssh start
+
```
Comment (weideng1) [inline Dockerfile:5]: normally you want to add some cleanup to avoid blowing up the image size

File: src/resources/runCommands.txt [hunk 4]
```diff
@@ -1,14 +1,14 @@
-// Download spark
-wget https://archive.apache.org/dist/spark/spark-2.4.8/spark-2.4.8-bin-hadoop2.6.tgz
-
-// Increase driver memory
---driver-memory 8G
+// Download dependencies
+curl -OL https://downloads.datastax.com/dsbulk/dsbulk.tar.gz
+curl -OL https://downloads.datastax.com/enterprise/cqlsh-astra.tar.gz
+wget https://archive.apache.org/dist/spark/spark-2.4.8/spark-2.4.8-bin-hadoop2.7.tgz
```
Comment (weideng1) [inline src/resources/runCommands.txt:4]: small nit: why we use wget instead of curl here?

File: src/resources/runCommands.txt [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
 
 // Migrate
-spark-submit --properties-file /<path>/sparkConf.properties --verbose --master "local[8]" --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-1.x.jar
+spark-submit --properties-file /<path>/sparkConf.properties --verbose --master "local[8]" --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-2.x.jar
+spark-submit --properties-file /<path>/sparkConf.properties --master "local[8]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-2.x.jar &> table_out.log
```
Comment (weideng1) [inline src/resources/runCommands.txt:8]: Is 25G here just an example? Did we already mention in the documentation that 32GB RAM or more is expected?

File: src/resources/runCommands.txt (hunk 4 above)
Comment (pravinbhat) [reply src/resources/runCommands.txt:4]: @weideng1 This file is not used, its just reference commands & we have both the formats (curl & wget)

File: src/resources/runCommands.txt (hunk 5 above)
Comment (pravinbhat) [reply src/resources/runCommands.txt:8]: Yes, this is just an example. Actual usage is based on use-case

**Review by weideng1 — APPROVED** [review]
LGTM

**Pull request merged**